PDSim/core/containers.pyx

```diff
@@ -547,15 +615,99 @@ cdef class ControlVolumeCollection(object):
         """
         cdef ControlVolume CV
         return [CV.State.get_dpdT() for CV in self.exists_CV]
+        
+    #TODO: missing xL, dudxL and the properties as mixture
+    """
+    For example:
+    
+    @property
+    def rho_m(self):
+        """
+            Density for each CV that exists
+        """
+        cdef ControlVolume CV
+        return [CV.StateFlood.get_rho_m() for CV in self.exists_CV]
+    
+    @property
+    def h_m(self):
+        """
+        Enthalpy for each CV that exists
+        """
+        cdef ControlVolume CV
+        return [CV.StateFlood.get_h_m() for CV in self.exists_CV]
+    
+    @property
+    def cp_m(self):
+        """
+        Specific heat at constant volume for each CV that exists
+        """
+        cdef ControlVolume CV
+        return [CV.StateFlood.get_cp_m() for CV in self.exists_CV]
+    
+    @property
+    def cv(self):
+        """
+        Specific heat at constant volume for each CV that exists
+        """
+        cdef ControlVolume CV
+        return [CV.StateFlood.get_cv_m() for CV in self.exists_CV]
+    
+    @property
+    def dpdT_m(self):
+        """
+        Derivative of pressure with respect to temperature at constant volume for each CV that exists
+        """
+        cdef ControlVolume CV
+        return [CV.StateFlood.get_dpdT_m() for CV in self.exists_CV]    
+    
+    @property
+    def xL(self):
+        """
+        Liquid mass fraction for each CV that exists
+        """
+        cdef ControlVolume CV
+        return [CV.StateFlood.get_xL() for CV in self.exists_CV]    
+
+    @property
+    def dudxL(self):
+        """
+        Liquid mass fraction for each CV that exists
+        """
+        cdef ControlVolume CV
+        return [CV.StateFlood.get_dudxL() for CV in self.exists_CV]
+    
+    
+    """
+    
+    
+    
+    #TODO: update the FloodStates
+    """
+    cpdef updateStates(self, str name1, arraym array1, str name2, arraym array2, str name3, arraym array3):
```

Comment:
A couple of options:

1) Overload the function so that the last two are optional
2) Create a new function that is called if ``__hasLiquid__`` is true